src/main/java/com/cburch/logisim/gui/log/Signal.java

```diff
@@ -176,7 +176,7 @@ public void resize(int newMaxSize) {
     if (newMaxSize == maxSize) return;
     if (newMaxSize == 0 || newMaxSize > maxSize) {
       // growing
-      if (firstIndex != 0) retainOnly(0, curSize, newMaxSize); // keeps all data, but shifts it left
+      retainOnly(0, curSize, newMaxSize); // keeps all data, but shifts it left
```

Comment:
This change is not obvious, please explain cause of the issue and how this change fixes it. Why is the if-statement not necessary anymore?